File: Pause/App/DebugLog.swift
import Foundation

/// 临时调试日志：写入 /tmp/pause_debug.log，用于定位设置页按钮交互问题。
/// 问题解决后可整体移除。
enum DebugLog {
    static let url = URL(fileURLWithPath: "/tmp/pause_debug.log")

    static func log(_ message: String) {
        let line = "\(Date()) [\(Thread.current.name ?? "main")] \(message)\n"
        let data = Data(line.utf8)
        if let handle = try? FileHandle(forWritingTo: url) {
            handle.seekToEndOfFile()
            handle.write(data)
            try? handle.close()
        } else {
            try? data.write(to: url)
        }
    }
}

File: src-tauri/src/debug_log.rs
//! 调试日志：对齐原 DebugLog.swift，追加写 pause_debug.log。
//! 默认静默，PAUSE_DEBUG=1 时启用，便于现场排查。

use std::fs::OpenOptions;
use std::io::Write;
use std::path::PathBuf;
use std::sync::Mutex;
use std::time::{SystemTime, UNIX_EPOCH};

static LOG_LOCK: Mutex<()> = Mutex::new(());

fn log_path() -> PathBuf {
    if cfg!(target_os = "windows") {
        std::env::temp_dir().join("pause_debug.log")
    } else {
        PathBuf::from("/tmp/pause_debug.log")
    }
}

/// append 一行时间戳日志；失败静默（日志绝不能影响主流程）。
pub fn debug_log(message: &str) {
    if std::env::var("PAUSE_DEBUG").ok().as_deref() != Some("1") {
        return;
    }
    let ms = SystemTime::now()
        .duration_since(UNIX_EPOCH)
        .map(|d| d.as_millis())
        .unwrap_or(0);
    let line = format!(
        "{ms}ms [{:?}] {message}\n",
        std::thread::current().name().unwrap_or("main")
    );
    let _guard = LOG_LOCK.lock();
    if let Ok(mut f) = OpenOptions::new()
        .create(true)
        .append(true)
        .open(log_path())
    {
        let _ = f.write_all(line.as_bytes());
    }
}
